network.go

```diff
@@ -322,6 +322,32 @@ func getInterface(netHandle *netlink.Handle, link netlink.Link) (*pb.Interface,
 	return &ifc, nil
 }
 
+func (s *sandbox) listInterfaces(netHandle *netlink.Handle) (*pb.Interfaces, error) {
+	if netHandle == nil {
+		netHandle, err := netlink.NewHandle(unix.NETLINK_ROUTE)
+		if err != nil {
+			return nil, err
+		}
+		defer netHandle.Delete()
+	}
+
+	links, err := netHandle.LinkList()
+	if err != nil {
+		return nil, err
+	}
+
+	var interfaces pb.Interfaces
+	for _, link := range links {
+		ifc, err := getInterface(netHandle, link)
+		if err != nil {
+			agentLog.WithError(err).Error("listInterfaces() failed")
```

Comment:
I'd be tempted to add a `.WithField("link", link)` here.

Reply:
Done